docs/contract-updatability.md

```diff
@@ -0,0 +1,340 @@
+Enums cases are represented using their raw-value at the Cadence interpreter and runtime.
+Hence, any change that causes an enum-case to change its raw value is not permitted.
+Otherwise, a changed raw-value could cause an already stored enum value to have a different meaning than what
+it originally was (type confusion).
+
+#### Valid Changes:
+- Adding an enum-case at the end of the existing enum-cases is valid.
+  ```cadence
+  // Existing enum
+
+  pub enum Color: Int {
+    pub case RED
+    pub case BLUE
+  }
+
+
+  // Updated enum
+
+  pub enum Color: Int {
+    pub case RED
+    pub case BLUE
+    pub case GREEN    // valid new enum-case at the bottom
+  }
+  ```
+#### Invalid Changes
+- Adding an enum-case at the top or in the middle of the existing enum-cases is invalid.
+  ```cadence
+  // Existing enum
+
+  pub enum Color: Int {
+    pub case RED
+    pub case BLUE
+  }
+
+
+  // Updated enum
+
+  pub enum Color: Int {
+    pub case RED
+    pub case GREEN    // invalid new enum-case in the middle
+    pub case BLUE
+  }
+  ```
+- Changing the name of an enum-case is invalid.
+  ```cadence
+  // Existing enum
+
+  pub enum Color: Int {
+    pub case RED
+    pub case BLUE
+  }
+
+
+  // Updated enum
+
+  pub enum Color: Int {
+    pub case RED
+    pub case GREEN    // invalid change of names
+  }
+  ```
+  - Previously stored raw values for `Color.BLUE` now represents `Color.GREEN`. i.e: The stored values has changed
+    their meaning, and hence not a valid change.
+  - Similarly, it is possible to add a new enum with the old name `BLUE`, which gets a new raw value. Then the same
+    enum-case `Color.BLUE` may have used two raw-values at runtime, before and after the change, which is also invalid.
+
+- Removing the enum case is invalid. Removing allows a one to add and remove an enum-case which has the same effect
+  as renaming.
+  ```cadence
+  // Existing enum
+
+  pub enum Color: Int {
+    pub case RED
+    pub case BLUE
+  }
+
+
+  // Updated enum
+
+  pub enum Color: Int {
+    pub case RED
+
+    // invalid removal of `case BLUE`
+  }
+  ```
+- Changing the order or enum-cases is permitted
+  ```cadence
+  // Existing enum
+
+  pub enum Color: Int {
+    pub case RED
+    pub case BLUE
+  }
+
+
+  // Updated enum
+
+  pub enum Color: UInt8 {
+    pub case BLUE   // invalid change of order
+    pub case RED
+  }
+  ```
+  - Raw value of an enum is implicit, and corresponds to the defined order.
+  - Changing the order of enum-cases has the same effect as changing the raw-value, which could cause the storage
+    inconsistencies and type-confusions as described earlier.
+
+## Functions
+Updating a function definition is always valid, as function definitions are never stored as data. 
+i.e: Function definition is a part of the code, but not data.
+- Changing a function signature (parameters, return types) is valid.
```

Comment:
Changing function signature is allowed at the moment. So I documented the current behavior.

Reply:
when you say signature, it also covers access modifiers? 

can I change access(all) to access(self) ?

Reply:
Yes. Added a note in 0e08133